crates/holochain/src/conductor/conductor.rs

```diff
@@ -1475,14 +1496,13 @@ mod app_impls {
             installed_app_id: InstalledAppId,
             manifest: AppManifest,
             agent_key: Option<AgentPubKey>,
-            defer_memproofs: bool,
             ops: AppRoleResolution,
-            ignore_genesis_failure: bool,
+            flags: InstallAppCommonFlags,
         ) -> ConductorResult<InstalledApp> {
             let dpki = self.running_services().dpki.clone();
 
             // if dpki is installed, load dpki state
-            let mut dpki = if let Some(d) = dpki.as_ref() {
+            let dpki = if let Some(d) = dpki.as_ref() {
```

Comment:
DPKI state is locked here until the end of the installation. Is that necessary? How about generally locking only when accessing it, e. g.
```rust
dpki_service.state().await.run_operation().await?;
```

Reply:
I did this when we were letting DPKI tell us the derivation details, in which case we don't want multiple concurrent installations getting into a race condition. I think maybe the lock is no longer necessary, but I'll have another think about it in the daytime.

Reply:
Any more thoughts? It seems there's nothing happening with DPKI between the state lock and when it is actually used to register the key.

Reply:
Yes, it's good you pointed this out, DPKI doesn't need to stay locked anymore. I removed the long lock,